packages/shylu/shylu_node/fastilu/src/shylu_fastilu.hpp

```diff
@@ -64,13 +64,172 @@
 #include <KokkosSparse_spmv.hpp>
 #include <KokkosSparse_sptrsv.hpp>
 #include <KokkosSparse_trsv.hpp>
+#include <KokkosSparse_BsrMatrix.hpp>
 #include <shylu_fastutil.hpp>
 
-//whether to print extra debug output at runtime to stdout
-//comment out next line to disable
+#include "Tpetra_BlockCrsMatrix_Helpers.hpp"
+
+// FASTILU Preprocesser options:
+
+// whether to print extra debug output at runtime to stdout
+// comment out next line to disable
 //#define FASTILU_DEBUG_OUTPUT
 
-//forward declaration
+// whether to print timings
+//#define FASTILU_TIMER
+
+// Whether to try to maintain exact behavior with unblocked impl. Comes at
+// a steep performance cost.
+//#define FASTILU_ONE_TO_ONE_UNBLOCKED
```

Comment:
This was the preprocessor setting I used for debugging. It forces fastilu to produce deterministic results such that block and unblocked impls produce the same result.